### PR TITLE
Fix FindGINKGO.cmake for Ginkgo 1.5.0

### PR DESCRIPTION
**Pull request by tcojean**
As the title says,

We are about to release Ginkgo 1.5.0. As part of that release, we introduced MPI. The current scheme of the `FindGINKGO.cmake` module expects all `ginkgo` dependencies to be `ginkgo` related only. With MPI, this is no longer true, so a simple check is added.

See https://github.com/spack/spack/pull/33824#issuecomment-1311043479 where this appeared first.

### CODE REVIEW

File: cmake/modules/FindGINKGO.cmake [hunk 1]
```diff
@@ -51,17 +51,19 @@ unset(GINKGO_CXX_COMPILER)
...
-    NAMES ${_library}
-    HINTS ${GINKGO_INSTALL_LIBRARY_DIR}
-    NO_DEFAULT_PATH
-    NO_CMAKE_ENVIRONMENT_PATH
-    NO_CMAKE_PATH
-    NO_SYSTEM_ENVIRONMENT_PATH
-    NO_CMAKE_SYSTEM_PATH
-    NO_CMAKE_FIND_ROOT_PATH
-    )
+  IF(_library MATCHES "ginkgo.*")
```
Comment (kronbichler) [inline cmake/modules/FindGINKGO.cmake:54]: This looks good to me, thank you. Would you mind adding a short comment about this:
```suggestion
  # Make sure to only pick up Ginkgo's own libraries here, skipping
  # the MPI libraries that are listed here as of Ginkgo 1.5.
  IF(_library MATCHES "ginkgo.*")
```

Comment (tcojean) [reply cmake/modules/FindGINKGO.cmake:56]: Thanks Martin. Good point!

**Review by kronbichler — APPROVED** [review]

**Comment by balay** [comment]
It would be great if this PR can be merged in quickly!

2 xsdk spack PRs are dependent on this PR getting merged.

https://github.com/spack/spack/pull/33838
https://github.com/spack/spack/pull/33794

Thanks!

**Comment by balay** [comment]
Thanks!

**Pull request merged**